frontend/packages/core/src/NPS/feedback.tsx

```diff
@@ -0,0 +1,223 @@
+      info: <MuiSuccessIcon style={{ color: "#3548d4" }} />,
+    },
+    style: {
+      margin: "32px",
+      alignItems: "center",
+    },
+  };
+
+  return (
+    <Alert severity="info" {...AlertProps}>
+      <Typography variant="subtitle3">Thank you for your feedback!</Typography>
+    </Alert>
+  );
+};
+
+/**
+ * NPS feedback component which is the base for both Wizard and Anytime.
+ * Will fetch given survey options from the server based on the given origin
+ * Then display a feedback component based on the given emoji ratings
+ *
+ * @param opts Available feedback options
+ * @returns NPSFeedback component
+ */
+const NPSFeedback = (opts: FeedbackOptions) => {
+  const [hasSubmit, setHasSubmit] = useState<boolean>(false);
+  const [selected, setSelected] = useState<Rating>(null);
+  const [freeformFeedback, setFreeformFeedback] = useState<string>("");
+  const [error, setError] = useState<boolean>(false);
+  const [survey, setSurvey] = useState<IClutch.feedback.v1.ISurvey>({});
+  const [requestId, setRequestId] = useState<string>("");
+  const maxLength = 180;
+  const debounceTimer = 500;
+
+  const trimmed =
+    freeformFeedback.trim().length > maxLength
+      ? `${freeformFeedback.trim().substring(0, maxLength - 3)}...`
+      : freeformFeedback;
+
+  const textFieldProps = {
+    fullWidth: true,
+    InputProps: {
+      rows: 3,
+      rowsMax: 3,
+    },
+  };
+
+  // Will fetch the survey results for the given origin on load
+  React.useEffect(() => {
+    let data: IClutch.feedback.v1.ISurvey = defaults;
+
+    client
+      .post("/v1/feedback/getSurveys", {
+        origins: [opts.origin],
+      })
+      .then(response => {
+        const surveyData: IClutch.feedback.v1.IGetSurveysResponse = response?.data?.originSurvey;
+
+        data = surveyData[opts.origin] ?? defaults;
+      })
+      .catch((err: ClutchError) => {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      })
+      .finally(() => {
+        setRequestId(uuid());
+        setSurvey(data);
+      });
+  }, []);
+
+  // Will debounce feedback requests to the server in case of multiple quick changes to selected
+  const sendFeedback = React.useCallback(
+    debounce((formData: IClutch.feedback.v1.ISubmitFeedbackRequest) => {
+      client
+        .post("/v1/feedback/submitFeedback", { userId: userId(), ...formData })
+        .catch((err: ClutchError) => {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        });
+    }, debounceTimer),
+    []
+  );
+
+  // On a change to submit or selected will attempt to send a feedback request
+  React.useEffect(() => {
+    if (selected) {
+      sendFeedback({
+        id: requestId,
+        feedback: {
+          ratingLabel: selected.label,
+          ratingScale: {
+            emoji: IClutch.feedback.v1.EmojiRating[selected.emoji],
+          },
+          urlPath: `${window.location.pathname}${window.location.search ?? ""}`,
```

Comment:
```suggestion
          urlPath: window.location.pathname,
```

synced offline. after the PR is merged, we'll open up a follow up to add the query params to the metadata object